source/common/http/header_map_impl.cc

```diff
@@ -17,195 +17,106 @@ namespace Envoy {
 namespace Http {
 
 namespace {
-constexpr size_t MinDynamicCapacity{32};
 // This includes the NULL (StringUtil::itoa technically only needs 21).
 constexpr size_t MaxIntegerLength{32};
 
-uint64_t newCapacity(uint32_t existing_capacity, uint32_t size_to_append) {
-  return (static_cast<uint64_t>(existing_capacity) + size_to_append) * 2;
-}
-
 void validateCapacity(uint64_t new_capacity) {
   // If the resizing will cause buffer overflow due to hitting uint32_t::max, an OOM is likely
   // imminent. Fast-fail rather than allow a buffer overflow attack (issue #1421)
   RELEASE_ASSERT(new_capacity <= std::numeric_limits<uint32_t>::max(),
                  "Trying to allocate overly large headers.");
-  ASSERT(new_capacity >= MinDynamicCapacity);
 }
 
-} // namespace
+absl::string_view
+get_str_view(const absl::variant<absl::string_view, absl::InlinedVector<char, 128>>& buffer) {
```

Comment:
nit: maybe add a using declaration to alias `absl::variant<absl::string_view, absl::InlinedVector<char, 128>>`? You could also sub alias `absl::InlinedVector<char, 128>`

Reply:
I used a macro before. Do you suggest a typedef or something else?

Reply:
`using InlineHeaderVector = absl::InlinedVector<char, 128>;`
`using VariantHeader = absl::variant<absl::string_view, InlineHeaderVector>;`

Etc.

Reply:
Will do. Thanks for the guidance.